collect_stata/write_json.py

```diff
@@ -149,128 +137,133 @@ def stats_number(elem, file_csv):
-        if sum(Counter(data_withoutmissings.values).values()) > 10:
-            meta_dict["statistics"] = uni_statistics(elem, file_csv)
-    else:
-        meta_dict["statistics"] = uni_statistics(elem, file_csv)
-
-    return meta_dict
-
-
 def generate_stat(data, metadata, study):
     """Prepare statistics for every variable
 
     Input:
-    data: pandas DataFrame (later called file_csv)
+    data: pandas DataFrame (later called data)
     metadata: dict (later called file_json)
     study: string
 
     Output:
     stat: OrderedDict
     """
 
-    stat = list()
-    elements = list()
-    for elem in metadata["resources"][0]["schema"]["fields"]:
-        elements.append(elem)
-    elements_length = len(elements)
-
-    for i, elem in enumerate(elements):
-        logging.info("%s/%s", str(i + 1), str(elements_length))
-        stat.append(stat_dict(elem, data, metadata, study))
+    for i, elem in enumerate(metadata):
+        logging.info("%s/%s", str(i + 1), str(len(metadata)))
+        metadata[i]["study"] = study
+        metadata[i].update({"statistics": uni_statistics(elem, data)})
+        if elem["scale"] == "cat":
+            metadata[i]["categories"].update(uni(elem, data))
 
-    return stat
+    return metadata
 
 
-def write_json(data, metadata, filename, study=""):
+def write_json(data, metadata, filename, study):
     """Main function to write json.
 
-    Input:
-    data: pandas DataFrame (later called file_csv)
-    metadata: dict (later called file_json)
-    filename: string
-    study: string
+    TODO: this function should be able to handel a list as metadata input.
+    This list will contain a dictionary for ever variable of a dataset.
+    Example:
+        .. code-block:: python
+
+            [
+                    {
+                        "name": "variable_name",
+                        "label": "variable_label",
+                        "type": "category",
+                        "scale": "cat",
+                        "categories": {
+                            "values": [-1, 1],
+                            "labels": ["[-1] invalid", "[1] valid"],
+                        },
+                    }
+            ]
+
+        metadata_new = [
+            {
+                "name": "HKIND",
+                "label": "Kinder",
+                "type": "category",
+                "scale": "cat",
+                "categories": {
+                    "values": [-1, 1, 2],
+                    "labels": ["keine Antwort", "Ja", "Nein"],
+                    "missings": [True, False, False],
+                },
+            },
+            {
+                "name": "HM04",
+                "label": "Miete in Euro",
+                "type": "int",
+                "scale": "number",
+            },
+            {
+                "name": "HKGEBA",
+                "label": "test for string var",
+                "type": "str",
+                "scale": "string",
+            }
+        ]
+
```

Comment:
If you are confident that the Todo is done, you can remove it. Otherwise it will open a new issue when this is merged.
I guess the example can remain but english labels would be more consistent.